### PR TITLE
[Instrument manager] Bug fix & cleanup

### PR DESCRIPTION
**Pull request by xlecours**
This pull request fix a bug when uploading an instrument in the instrument manager.

Argument 2 and 3 of `\NDB_BVL_Instrument::factory` must be strings
Also fix `PHP Notice:  Only variables should be assigned by reference`

See also: RM#13643 & RM#13644

### CODE REVIEW

File: modules/instrument_manager/test/TestPlan.md [hunk 1]
```diff
@@ -1,12 +1,10 @@
...
 
 1. User has access to Instrument Manager if he/she has permission “superuser”.
    [Automation Testing]
-2. Check that tab exists on a default install (for superusers).
+2. Check that `Instrument Manager` menu-item exists on a default install under Admin menu-item (for superusers).
    [Automation Testing]
-3. Check that the warning text above is accurate for both with and without the directory writable by Apache. The select file/upload button should only be there if it's writable.
-   [Manual Testing]
-4. Verify GUI is updated, menu tabs are working.
+3. Check that the warning text is accurate if the `project/instruments` or `project/sql_tables` directories are not writable by Apache. The select file/upload panel should only be there if it's writable.
```
Comment (driusan) [inline modules/instrument_manager/test/TestPlan.md]: Point 5 below is misnumbered now.. and I don't know if it makes sense for point `2` to be part of the test plan. The LORIS menu isn't really part of the module.

**Comment by xlecours** [comment]
I accidentally pushed in the same branch... I figured that it was related so I kept it like this.

File: modules/instrument_manager/test/TestPlan.md (hunk 1 above)
Comment (driusan) [reply modules/instrument_manager/test/TestPlan.md]: Though I guess (other than the misnumbering) that's also outside of the scope of this PR.

Comment (xlecours) [reply modules/instrument_manager/test/TestPlan.md]: I agree for the menu except for the fact that it is the usual entry point. Do we have a LORIS Menu test suit?

Comment (driusan) [reply modules/instrument_manager/test/TestPlan.md]: No, but we probably should.

**Pull request merged**